src/main/java/seedu/address/logic/parser/ClientNoteAddCommandParser.java

```diff
@@ -25,20 +27,28 @@
     public ClientNoteAddCommand parse(String userInput) throws ParseException {
         requireNonNull(userInput);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(userInput, /* PREFIX_TAG,*/ PREFIX_NOTE);
+                ArgumentTokenizer.tokenize(userInput, PREFIX_TAG, PREFIX_NOTE);
         if (!arePrefixesPresent(argMultimap, PREFIX_NOTE) || argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ClientNoteAddCommand.MESSAGE_USAGE));
         }
         Index index;
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            String[] preambleWords = argMultimap.getPreamble().split(" "); // todo: ask if there's a better way?
+            index = ParserUtil.parseIndex(preambleWords[preambleWords.length - 1]);
         } catch (ParseException pe) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                     ClientNoteAddCommand.MESSAGE_USAGE), pe);
         }
+        Tag tag;
+        try {
+            tag = ParserUtil.parseTag(argMultimap.getValue(PREFIX_TAG).orElse("untagged"));
+        } catch (ParseException pe) {
+            tag = ParserUtil.parseTag("untagged"); // tagging a note should be optional
+        }
         Note clientNote = ParserUtil.parseNote(argMultimap.getValue(PREFIX_NOTE)
                 .orElseThrow(() -> new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                         ClientNoteAddCommand.MESSAGE_USAGE))));
+        clientNote.addTag(tag);
```

Comment:
Should `untagged` tags be added to the note?

Reply:
I think this is useful. Then can filter by untagged notes and do other actions based off that.